### DISCUSSION TITLE
Native AOT

### DISCUSSION THREAD

**Discussion by Daniel15** [picoe/Eto#2659] [General]
Does Eto support [Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/)? I'm looking at different UI frameworks that support Linux, but I'd like to be able to compile my app into a single, statically-linked executable file to make distribution easier.

**Comment by cwensley** [comment]
Hey @Daniel15,

Thanks for the inquiry.  There has not been any work involving AOT with Eto, so likely some of the reflection and dynamic member access would have problems, but I haven't tested it.  I have worked on adding the attributes that would be necessary for this to be better supported, but likely there'd be certain features that wouldn't work (PropertyGrid/PropertyCell, for example as it fully works on reflection)

This doesn't mean you can't distribute your app as a (non AOT) single file executable. You can usually trim all assemblies except Eto and your UI code which does produce a relatively small footprint.

Hope this helps.

**Comment by ris-work** [comment]
> Does Eto support [Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/)? I'm looking at different UI frameworks that support Linux, but I'd like to be able to compile my app into a single, statically-linked executable file to make distribution easier.

I am also looking forward for the AOT compilation support. I can confirm that AOT compilation does neither work with WPF nor WinForms `hello-world` Eto application. Eto apps are still very responsive without AOT, though (did not notice much performance differences).

**Comment by trypsynth** [comment]
I too am waiting for Native AOT support in ETO, it's an absolutely amazing framework and being able to give my users a lean-and-mean, fully standalone binary with an ETO GUI in it would be incredible!
